e2e/app/HomePage.ts

```diff
@@ -0,0 +1,55 @@
+import {ElementHandle, Page} from 'puppeteer';
+import {waitUntilFindTexts} from '../driver/waitFuncs';
+import Link from './aou-elements/Link';
+import {findIcon} from './aou-elements/xpath-finder';
+import AuthenticatedPage from './page-mixin/AuthenticatedPage';
+import {SideNav} from './page-mixin/SideNav';
+
+const configs = require('../resources/workbench-config.js');
+
+export const FIELD_LABEL = {
+  TITLE: 'Homepage',
+  HEADER: 'Workspaces',
+  SEE_ALL_WORKSPACES: 'See all Workspaces',
+  CREATE_NEW_WORKSPACE: 'Workspaces',
+};
+
+
+class Home extends AuthenticatedPage {
+
+  public page: Page;
+  constructor(page: Page) {
+    super(page);
+    this.page = page;
+  }
+
+  public async isLoaded(): Promise<boolean> {
+    await super.isLoaded(FIELD_LABEL.TITLE);
+    await new Link(this.puppeteerPage).withLabel(FIELD_LABEL.SEE_ALL_WORKSPACES);
+    await waitUntilFindTexts(this.puppeteerPage, FIELD_LABEL.HEADER);
+    return true;
```

Comment:
I'm not sure that this method really fits in alongside the other methods. I *think* the purpose here is to wait for the page to fully load, but the name "isLoaded" suggests that it would be checking whether the page is *currently* loaded – which is definitely not the case.

For example: is there ever any scenario where this.isLoaded will resolve with a false value? If not, then I think it's meaningless to be resolving with a boolean at all.

Proposal: rename to waitForLoad, for more accurate name and consistency with other methods on the BasePage subclasses.

Reply:
`isLoaded` and `waitForReady` serve different purposes. `isLoaded` returns true if target page is indeed the right DOM, doesn't care about spinner or readiness. Updated `isLoaded` and `waitForReady`.

Reply:
forgot, but I shall rename `waitForReady` to `waitForLoad` in next commit while I address new feedback on latest commit.